### PR TITLE
unify retrieval names

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

preserve consistency between functional and class-like python modules, part of #729 

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: torchmetrics/retrieval/hit_rate.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 from torch import Tensor, tensor
 
 from torchmetrics.functional.retrieval.hit_rate import retrieval_hit_rate
-from torchmetrics.retrieval.retrieval_metric import RetrievalMetric
+from torchmetrics.retrieval.metric_base import RetrievalMetric
```
Comment (rohitgr7) [inline torchmetrics/retrieval/hit_rate.py:19]: how about just base?

**Review by rohitgr7 — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley:

File: tests/retrieval/test_mrr.py [hunk 2]
```diff
@@ -28,7 +28,7 @@
     _errors_test_functional_metric_parameters_default,
 )
 from torchmetrics.functional.retrieval.reciprocal_rank import retrieval_reciprocal_rank
-from torchmetrics.retrieval.mean_reciprocal_rank import RetrievalMRR
+from torchmetrics.retrieval.rreciprocal_rank import RetrievalMRR
```
Comment (lucadiliello) [inline tests/retrieval/test_mrr.py:31]: Why did you add an `r` at the beginning of the name?

Comment (Borda) [reply tests/retrieval/test_mrr.py:31]: Good catch 👍

**Review by lucadiliello — APPROVED** [review]
